include/api.php

```diff
@@ -5905,19 +5905,25 @@ function api_friendica_notification($type)
 	if ($a->argc!==3) {
 		throw new BadRequestException("Invalid argument count");
 	}
-	$notes = DI::notification()->getAll([], ['seen' => 'ASC', 'date' => 'DESC'], 50);
+
+	$notifications = DI::notification()->select([], ['order' => ['seen' => 'ASC', 'date' => 'DESC'], 'limit' => 50]);
 
 	if ($type == "xml") {
-		$xmlnotes = [];
-		if (!empty($notes)) {
-			foreach ($notes as $note) {
-				$xmlnotes[] = ["@attributes" => $note];
+		$xmlnotes = false;
+		if (!empty($notifications)) {
+			foreach ($notifications as $notification) {
+				$xmlnotes[] = ["@attributes" => $notification->toArray()];
 			}
 		}
 
-		$notes = $xmlnotes;
+		$result = $xmlnotes;
+	} elseif (count($notifications) > 0) {
+		$result = $notifications->getArrayCopy();
+	} else {
+		$result = false;
```

Comment:
Why change the result type if there's no result?

Reply:
A test failed because it asserts that an "empty" result is returned as array `['note' => false]`.
With the current behavior, it would be `['note' => []]`. As I don't know how if there are sideeffects for consumers, I tried to rework the behavior like asserted

Reply:
Ugh, ok.